xaitk_saliency/impls/vis_sal_classifier/occlusion_scoring.py

```diff
@@ -0,0 +1,51 @@
+from xaitk_saliency import ImageClassifierSaliencyMapGenerator
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+
+import numpy as np
+
+
+class OcclusionScoring (ImageClassifierSaliencyMapGenerator):
+    """
+    This saliency implementation transforms black-box
+    image classification scores into saliency heatmaps. This should
+    require a sequence of per-class confidences predicted on the
+    reference image, a number of per-class confidences as predicted
+    on perturbed images, as well as the masks of the reference image
+    perturbations (as would be output from a
+    `PerturbImage` implementation.
+    """
+
+    def generate(
+            self,
+            image_conf: np.ndarray,
+            perturbed_conf: np.ndarray,
+            perturbed_masks: np.ndarray
+    ) -> np.ndarray:
+
+        if not (perturbed_masks[0][0][0] in [0, 1]):
+            raise ValueError("Image perturbation mask must be of",
```

Comment:
Maybe OK for now, but this may eventually cause issues with other perturbation algorithms that output float-type masks (e.g. RISE). It may be better for this implementation to document a rule in how it handles float-type masks (e.g. rounds valuesl, then binarizes).